Reapply "DELWAQ-1199: Replace TimMultBl by an absolute time step"

HiddeElzinga opened the pull request:
This reverts commit b26624f4bc399a5ecde715eca6a33991ea8676bc.

# What was done 

<a short description with bullets> 

- e.g. Restarts are made more robust 
- e.g. Fixes a bug related to the writing of water depth on the map file 
- e.g. Introduces a new functionality on energy losses at bridge peirs 
- e.g. … 
 

# Evidence of the work done 

- [ ]	Video/figures \
<add video/figures if applicable> 
- [ ]	Clear from the issue description 
- [ ]	Not applicable 

# Tests 
- [ ] Tests updated \
<add testcase numbers if applicable, Issue number>
- [ ]	Not applicable 

# Documentation  
- [ ]	Documentation updated \
<add description of changes if applicable, Issue number> 
- [ ]	Not applicable 

# Issue link

HiddeElzinga committed:
Reapply "DELWAQ-1199: Replace TimMultBl by an absolute time step"

This reverts commit b26624f4bc399a5ecde715eca6a33991ea8676bc.

ye4deltares reviewed (approved):
updated code seems ok.

ye4deltares committed:
Merge branch 'main' into all/bugfix/DELWAQ-1199-replace-timmultbl-to-be-merged

ye4deltares committed:
Update sub file in the testcases

arjenmarkusdeltaresnl committed:
Merge branch 'main' into all/bugfix/DELWAQ-1199-replace-timmultbl-to-be-merged

arjenmarkusdeltaresnl committed:
Correct status of istep and initial value

It seems that the recent test failures might be due to the "phasing" of the calculations. A closer look at the changes for BLOOM timesteps and the things that remained the same, brought this to light.

arjenmarkusdeltaresnl committed:
DELWAQ-1199: Adjust the time interval for calling BLOOM

After some further consideration (only the BLOOM test cases were failing) I have adjusted the interval within which BLOOM is to be called. The tricky thing is that the time step in D-Flow FM is a floating-point number. This means care needs to be taken regarding round-offs.

arjenmarkusdeltaresnl committed:
DELWAQ-1199: Updates to various test cases

Time stamps changed, because of change from TimMultBl to BloomStep.

ye4deltares committed:
Merge branch 'main' into all/bugfix/DELWAQ-1199-replace-timmultbl-to-be-merged

arjenmarkusdeltaresnl committed:
Merge branch 'all/bugfix/DELWAQ-1199-replace-timmultbl-to-be-merged' of https://github.com/Deltares/Delft3D into all/bugfix/DELWAQ-1199-replace-timmultbl-to-be-merged

arjenmarkusdeltaresnl committed:
DELWAQ-1199: Adjust condition for BLOOM and adjust timestamp for c043

Hopefully the final touches: a safer condition for calling BLOOM and an update for the reference files of case e02_f030_c043.

ye4deltares committed:
Merge branch 'main' into all/bugfix/DELWAQ-1199-replace-timmultbl-to-be-merged
